pyproject.toml

```diff
@@ -141,6 +141,12 @@ ignore = [
     # Manual dict comprehension.
     "PERF403",
 
+    # Convert Dict[X] to dict[x], which we cannot do while we support Python 3.8
+    "UP006",
```

Comment:
For this, and the next one, shouldn't the target version already be skipping these? That's generally better than manually maintaining the upgrade list.

Reply:
It should, but it doesn't.

Let me double-check, I may have left this line when I was trying to upgrade `ruff` at the same time.

Reply:
Hmm, [looks like](https://docs.astral.sh/ruff/settings/#target-version) we should remove target-version in favour of our Python version specification.

Reply:
Hmm, [looks like](https://docs.astral.sh/ruff/settings/#target-version) we should remove target-version in favour of our Python version specification.

Reply:
Right, so I was getting a tox lint warning in `ops/pebble.py` which I couldn't figure a good way around.

My solution was to enable `from __future__ import annotations` and update all the type hints. A huge mechanical change. Somehow ruff then applies these two rules that surely should be gated by py 3.10+, and normally they are, but not when `__future__` is enabled.

I suppose I could try to fix that original single linting issue some other way, maybe with I could simply suppress it?

Reply:
Or maybe we should be setting [keep-runtime-typing](https://docs.astral.sh/ruff/settings/#lint_pyupgrade_keep-runtime-typing)?

I'm in favour of having the future import (until it's no longer required because it's the standard behaviour in the earliest version we support). It would be nice if `ruff` still applied the Python version around the pyupgrade checks, so we don't need to remember to go remove those when we move to a newer Python. But it's not that bad to need to do it.